src/Microsoft.AspNetCore.Mvc.Razor.Host/MvcRazorParser.cs

```diff
@@ -87,29 +87,7 @@ public class MvcRazorParser : RazorParser
                 _viewImportsDirectiveDescriptors,
                 errorSink);
 
-            var descriptors = visitor.GetDescriptors(documentRoot);
-            foreach (var descriptor in descriptors)
-            {
-                foreach (var attributeDescriptor in descriptor.Attributes)
-                {
-                    if (attributeDescriptor.IsIndexer &&
-                        string.Equals(
-                            attributeDescriptor.TypeName,
-                            _modelExpressionTypeName,
-                            StringComparison.Ordinal))
-                    {
-                        errorSink.OnError(
-                            SourceLocation.Undefined,
-                            Resources.FormatMvcRazorParser_InvalidPropertyType(
-                                descriptor.TypeName,
-                                attributeDescriptor.Name,
-                                _modelExpressionTypeName),
```

Comment:
Missed the second part here.

This is a `public` type and we cannot make breaking changes at this point. Also not worth adding another constructor overload without the unnecessary `modelExpressionTypeName` parameter. So, leave the constructor parameter and the `null` check alone. Just remove the `_modelExpressionTypeName` field -- as you've done.